code/src/hooks/use-filters-with-apply/useFiltersWithApply.test.ts

```diff
@@ -0,0 +1,160 @@
+import { act, renderHook } from "@testing-library/react";
+import { useSearchParams } from "react-router-dom";
+
+import parseFiltersFromParams from "@/hooks/use-filters-with-apply/parse-filters-from-params/parseFiltersFromParams";
+import serializeToQueryString from "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString";
+import useFiltersWithApply from "@/hooks/use-filters-with-apply/useFiltersWithApply";
+
+jest.mock("react-router-dom", () => ({
+  useSearchParams: jest.fn()
+}));
+
+jest.mock(
+  "@/hooks/use-filters-with-apply/parse-filters-from-params/parseFiltersFromParams"
+);
+jest.mock(
+  "@/hooks/use-filters-with-apply/serialize-to-query-string/serializeToQueryString"
+);
+
+const mockSetSearchParams = jest.fn();
+
+describe("useFiltersWithApply", () => {
+  let defaultFilters: Record<string, unknown>;
```

Comment:
Is there a need to create this variable here and than set value in `beforeEach`? Mabe create it as const outside of describe with the value that settled in `beforeEach`?

Reply:
maybe also define with `const` once be better. Because this variable has rewritten only once

Reply:
Moved defaultFilters to constant variable outside describe block as follows:

```tsx
const defaultFilters = { filter1: "default1", filter2: "default2" };
```